File: cpp/problem2.cpp
#include <iostream>

using namespace std;

int main() {
    int sum = 0;
    for(int x = 1, y = 1; x < 4000000;) {
        if(x%2 == 0)
            sum += x;

        int temp = x;
        x = x+y;
        y = temp;
    }

    cout << sum;
}

File: python/problem2.py
from itertools import takewhile

def fib():
    prev, curr = 1, 1
    while True:
        yield curr
        t, curr = curr, curr + prev
        prev = t

def solution():
    bounded_fib = takewhile(lambda x: x < 4 * 10 ** 6, fib()) 
    return sum(x for x in bounded_fib if x % 2 == 0)